lib/manager/components/deployment/CustomFileEditor.js

```diff
@@ -40,11 +40,37 @@ export default class CustomFileEditor extends Component<Props, {
 
   _canEdit = () => this.props.customFileEditIdx === null
 
-  _isEditing= () => this.props.idx === this.props.customFileEditIdx
+  /**
+   * Makes sure that the filename is valid. The filename is optional when a uri
+   * is provided, but is required when entering raw input.
+   */
+  _fileNameValid = (): boolean => {
+    const {fileSource, model: customFile} = this.state
+    return (fileSource === 'uri' && !!customFile.uri) ||
+      (!!customFile.contents && !!customFile.filename)
```

Comment:
Does this check for URI belong here? It seems like a simpler approach would just be:
```suggestion
    const {contents, filename} = customFile
    return fileSource === 'raw' && Boolean(contents) && Boolean(filename)
```

Reply:
Yes, it's required. The proposed change would return `false` for any custom file with a URI which is not correct since a filename is optional when a URI is provided.

Reply:
After looking at this more closely I ended up rewriting this so that it always returns `true` for a custom file with a URI. See 00445ee33e7b02c77a650d4a4098e64b6c528c67